resource-service/README.md

```diff
@@ -18,46 +18,85 @@ limitations under the License.
+
+### [AkkaRunnableGraph](src/main/java/uk/gov/gchq/palisade/service/resource/stream/config/AkkaRunnableGraph.java)
+The Resource Service accepts an incoming message from the `user` Kafka topic as processed by the [User Service](../user-service) which contains the `resourceId` that is being accessed. 
+This `resourceId` could be an actual file, or a directory that could contain many files and/or sub-directories. 
+The `resourceId` is then used as a query returning all `LeafResource`s under that path (eg. this may be all files in all (sub-)directories).
+All the returned resources will be within an Akka stream and each element in the stream is added to the `resource` Kafka topic to be processed by the [Policy Service](../policy-service). 
+If an exception was thrown, it is written to the `error` Kafka topic, along with suitable identifying metadata. 
+
+### [ResourceServicePersistenceProxy](src/main/java/uk/gov/gchq/palisade/service/resource/service/ResourceServicePersistenceProxy.java)
+The service will then query the 'cache' backing store to see if the requested `resourceId` has been stored. 
+A successful response from this implementation will then be added to the backing store, but errors are ignored. 
+If this is not the case then the request will be passed onto the local implementation of the Resource Service. 
+
+### ResourceService implementation, eg. [SimpleResourceService](src/main/java/uk/gov/gchq/palisade/service/resource/service/SimpleResourceService.java)
+The `resourceId` is treated as a file path in this case, and the returned resources come from Java's `Files::walk`. 
+
+
+## Kafka Message Model
 
 | ResourceRequest | ResourceResponse | AuditErrorMessage | 
-|:----------------|:-----------------|:------------------|
-| userId          | userId           | *token            | 
-| resourceId      | resourceId       | userId            |  
-| context         | context          | resourceId        |
-| user            | user             | context           | 
-|                 | resource         | exception         | 
-|                 |                  | serverMetadata    | 
+|:----------------|:-----------------|:------------------| 
+| *token          | *token           | *token            | 
+|  userId         |  userId          |  userId           | 
+|  resourceId     |  resourceId      |  resourceId       | 
+|  context        |  context         |  context          | 
+|  user           |  user            |  exception        | 
+|                 |  resource        |  serverMetadata   | 
+
+(fields marked with `*` are acquired from headers metadata)
 
-(fields marked with * are acquired from headers metadata)
+The service accepts a `ResourceRequest` from the User service and finds all the resources associated with the resourceId as a stream. 
+The stream is then consumed and for each resource a `ResourceResponse` is created and then sent to the Policy service for further processing. 
```

Comment:
Worth replacing all ResourceResponses with AuditableResourceResponse, see the Palisade Service readme

Reply:
The service in general produces a `ResourceResponse`, and only an `AuditErrorMessage` in exceptional circumstances.
I think it'd be better to focus on 'expected behaviour' than muddying the waters with exception handling here